components/collector/tests/base_collectors/test_collector.py

```diff
@@ -178,26 +181,22 @@ async def test_fetch_with_post_error(self):
             {"has_error": False, "sources": [self._source()], "metric_uuid": "metric_uuid", "report_uuid": "report1"},
         )
 
-    @patch("asyncio.sleep", Mock(side_effect=RuntimeError))
+    @patch("asyncio.sleep", AsyncMock(side_effect=[RuntimeError]))
     @patch("builtins.open", mock_open())
     async def test_collect(self):
         """Test the collect method."""
         mock_async_get_request = AsyncMock()
         mock_async_get_request.json.side_effect = [self.pip_json]
         with (
             self._patched_get(mock_async_get_request),
-            patch(
-                "quality_time_collector.database_connection",
-                return_value=self.database,
-            ),
-            patch(self.create_measurement) as post,
+            patch("quality_time_collector.database_connection", return_value=self.database),
             self.assertRaises(RuntimeError),
         ):
             await quality_time_collector.collect()
-        post.assert_called_once_with(
-            self.database,
-            {"has_error": False, "sources": [self._source()], "metric_uuid": "metric_uuid", "report_uuid": "report1"},
-        )
+        # Wait for the created tasks to finish
+        await asyncio.gather(*asyncio.all_tasks() - {asyncio.current_task()})
+        measurement = next(self.database["measurements"].find({"metric_uuid": "metric_uuid"}))
+        self.assertEqual("metric_uuid", measurement["metric_uuid"])
```

Comment:
Why assert that the found object is as queried, shouldn't `next()` already `raise StopIteration` if it doesn't match?

Reply:
Correct. I'll change the assert to check the connection error that happens due to the session being closed immediately.